DisCatSharp/Clients/DiscordClient.Dispatch.cs

```diff
@@ -1353,26 +1368,72 @@ internal async Task OnGuildScheduledEventCreateEventAsync(DiscordEvent scheduled
+                old.Description = scheduled_event.Description;
+                old.ChannelId = scheduled_event.ChannelId;
+                old.EntityMetadata = scheduled_event.EntityMetadata;
+                old.EntityType = scheduled_event.EntityType;
+                old.Status = scheduled_event.Status;
+                old.ScheduledStartTimeRaw = scheduled_event.ScheduledStartTimeRaw;
+                old.ScheduledEndTimeRaw = scheduled_event.ScheduledEndTimeRaw;
+                old.PrivacyLevel = scheduled_event.PrivacyLevel;
+                old.UserCount = scheduled_event.UserCount;
+                //old.SkuIds = scheduled_event.SkuIds;
+                return old;
+            });
 
-            await this._guildScheduledEventUpdated.InvokeAsync(this, new GuildScheduledEventUpdateEventArgs(this.ServiceProvider) { ScheduledEvent = scheduled_event, Guild = scheduled_event.Guild }).ConfigureAwait(false);
+            await this._guildScheduledEventUpdated.InvokeAsync(this, new GuildScheduledEventUpdateEventArgs(this.ServiceProvider) { ScheduledEventBefore = old_event, ScheduledEventAfter = scheduled_event, Guild = scheduled_event.Guild }).ConfigureAwait(false);
         }
 
         /// <summary>
         /// Dispatches the <see cref="GuildScheduledEventDeleted"/> event.
         /// </summary>
         /// <param name="scheduled_event">The deleted event.</param>
-        internal async Task OnGuildScheduledEventDeleteEventAsync(DiscordEvent scheduled_event)
+        internal async Task OnGuildScheduledEventDeleteEventAsync(DiscordScheduledEvent scheduled_event)
         {
             scheduled_event.Discord = this;
             var guild = this.InternalGetCachedGuild(scheduled_event.GuildId);
-            guild._scheduledEvents[scheduled_event.Id] = scheduled_event;
 
-            await this._guildScheduledEventDeleted.InvokeAsync(this, new GuildScheduledEventDeleteEventArgs(this.ServiceProvider) { ScheduledEvent = scheduled_event, Guild = scheduled_event.Guild }).ConfigureAwait(false);
+            guild._scheduledEvents.TryRemove(scheduled_event.Id, out var deleted_event);
+
+            await this._guildScheduledEventDeleted.InvokeAsync(this, new GuildScheduledEventDeleteEventArgs(this.ServiceProvider) { ScheduledEvent = scheduled_event, Guild = scheduled_event.Guild, Status = scheduled_event.Status }).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Dispatches the <see cref="GuildScheduledEventUserAdded"/> event.
+        /// </summary>
+        internal async Task OnGuildScheduledEventUserAddedEventAsync(ulong guild_scheduled_event_id, ulong user_id, ulong guild_id)
+        {
+            var guild = this.InternalGetCachedGuild(guild_id);
+            var success = guild._scheduledEvents.TryGetValue(guild_scheduled_event_id, out var scheduled_event);
+            if (!success)
+                scheduled_event = await this.ApiClient.GetGuildScheduledEventAsync(guild_id, guild_scheduled_event_id);
```

Comment:
This is completely asinine. I wouldn't advise making hidden API calls like this.

Reply:
Yeah, I'm working on fixing it right now.
Thanks for the advise